File: backend/src/main/java/com/nilanjan/backend/appointment/domain/Appointment.java
package com.nilanjan.backend.appointment.domain;

import java.time.Instant;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "appointments")
public class Appointment {

    @Id
    private ObjectId id;

    private String appointmentCode;

    private ObjectId patientId;
    private ObjectId doctorId;

    private Instant scheduledStart;
    private Instant scheduledEnd;

    private String reason;

    private AppointmentStatus status;

    private ObjectId createdBy;
    private Instant createdAt;

    private Instant checkedInAt;
    private Instant startedAt;
    private Instant completedAt;
    private Instant cancelledAt;
    private String cancelledByRole;
    private String cancelledReason;
}

File: frontend/types/appointment.ts
export type AppointmentStatus =
  | "SCHEDULED"
  | "CHECKED_IN"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "CANCELLED"
  | "NO_SHOW";

export interface AppointmentResponse {
  id: string;
  appointmentCode: string;
  patientId: string;
  doctorId: string;
  scheduledStart: string;
  scheduledEnd: string;
  status: AppointmentStatus;
}

export interface CreateAppointmentRequest {
  patientId: string;
  doctorId: string;
  scheduledStart: string;
  scheduledEnd: string;
  reason: string;
}
